usr/src/lib/libict_pymod/ict.py

```diff
@@ -2517,19 +2524,20 @@ def setup_sudo(self, login):
         temp_file = '/var/run/sudoers'
 
         try:
-            with open(self.sudoers, 'r') as fp:
-                sudoers_lines = fp.readlines()
-            with open(temp_file, 'w') as fp_tmp:
-                for l in sudoers_lines:
-                    if not l.startswith("jack"):
-                        fp_tmp.write(l)
-                if login:
-                    fp_tmp.write(login + ' ALL=(ALL) ALL\n')
-
-            os.remove(self.sudoers)
-            shutil.move(temp_file, self.sudoers)
-            os.chmod(self.sudoers, S_IREAD | S_IRGRP)
-            os.chown(self.sudoers, 0, 0) # chown root:root
+            if os.path.isfile(self.sudoers):
```

Comment:
Generally, in Python it is a practice to not do things like if os.path.isfile, but try the action directly, e.g. open. It's considered to be more Pythonic. It is also considered an anti-pattern: http://docs.quantifiedcode.com/python-anti-patterns/readability/asking_for_permission_instead_of_forgiveness_when_working_with_files.html